.github/workflows/ci-cd.yml

```diff
@@ -41,20 +41,15 @@ jobs:
         os:
         - ubuntu-24.04
         - ubuntu-24.04-arm
-        include:
-        - experimental: false
-        - upload-coverage: false
-        - upload-coverage: true
-          python-version: 3.11
-          os: ubuntu-24.04
       fail-fast: false
     uses: ./.github/workflows/reusable-test.yml
     with:
       python-version: ${{ matrix.python-version }}
       os: ${{ matrix.os }}
-      continue-on-error: ${{ matrix.experimental }}
+      continue-on-error: false
       enable-cache: ${{ github.ref_type == 'tag' && 'false' || 'auto' }}
-      upload-coverage: ${{ matrix.upload-coverage }}
+      upload-coverage:
+        ${{ matrix.python-version == 3.11 && matrix.os == 'ubuntu-24.04' }}
```

Comment:
Why have it conditional, even?

Reply:
and why `3.11` in particular :p 

proper solution would indeed be flags+uploads on every matrix entry, but for this PR I just wanted to pull out this change from #1085 so it's less of a pain to keep that PR up to date.